pkg/kubelet/dockertools/docker.go

```diff
@@ -625,4 +729,6 @@ func parseImageName(image string) (string, string) {
 
 type ContainerCommandRunner interface {
 	RunInContainer(containerID string, cmd []string) ([]byte, error)
+	ExecInContainer(containerID string, cmd []string, in io.Reader, out, err io.WriteCloser, tty bool) error
+	PortForward(containerID string, port uint16, stream io.ReadWriteCloser) error
```

Comment:
Ultimately we need to be able to easily extract this whole function from the kubelet and run it alongside as a standalone process.  Does the Server class then support other function like log streaming that goes with it?  Seems not unreasonable to think of it as kubelet-server process.


> On Jan 29, 2015, at 4:30 PM, Andy Goldstein <notifications@github.com> wrote:
> 
> In pkg/kubelet/dockertools/docker.go:
> 
> > @@ -625,4 +729,6 @@ func parseImageName(image string) (string, string) {
> >  
> >  type ContainerCommandRunner interface {
> >  	RunInContainer(containerID string, cmd []string) ([]byte, error)
> > +	ExecInContainer(containerID string, cmd []string, in io.Reader, out, err io.WriteCloser, tty bool) error
> > +	PortForward(containerID string, port uint16, stream io.ReadWriteCloser) error
> Should I move this to a new ContainerPortForwarder interface and add a portForwarder field to the Kubelet?
> 
> —
> Reply to this email directly or view it on GitHub.
>